File: Csharp/Models/Comment.cs
using System.ComponentModel.DataAnnotations;

namespace BlogSystem.Models;

public class Comment
{
    public int Id { get; set; }
    
    [Required]
    public string Content { get; set; } = string.Empty;
    
    public DateTime CreatedAt { get; set; } = DateTime.Now;
    
    public int UserId { get; set; }
    public User User { get; set; } = null!;
    
    public int PostId { get; set; }
    public Post Post { get; set; } = null!;
}

File: Node/src/models/Comment.ts
import { User } from './User';
import { Post } from './Post';

export interface Comment {
  id: number;
  content: string;
  createdAt: Date;
  userId: number;
  postId: number;
  user?: User;
  post?: Post;
}

export interface CreateCommentRequest {
  content: string;
  userId: number;
  postId: number;
}
